onadata/apps/main/management/commands/update_enketo_urls.py

```diff
@@ -45,25 +48,45 @@ def handle(self, *args, **options):
 
         resultset = MetaData.objects.filter(
             Q(data_type='enketo_url') | Q(data_type='enketo_preview_url'))
-        for meta_data in resultset:
-            username = meta_data.content_object.user.username
-            id_string = meta_data.content_object.id_string
-            data_type = meta_data.data_type
-            data_value = meta_data.data_value
-            xform = meta_data.content_object
-            with open('/tmp/enketo_url', 'a') as f:
-
-                if data_type == 'enketo_url':
-                    form_url = get_form_url(
-                        request, username, protocol=protocol,
-                        xform_pk=xform.pk)
-                    _enketo_url = enketo_url(form_url, id_string)
-                    MetaData.enketo_url(xform, _enketo_url)
-                elif data_type == 'enketo_preview_url':
-                    _enketo_preview_url = get_enketo_preview_url(
-                        request, username, id_string, xform_pk=xform.pk)
-                    MetaData.enketo_preview_url(xform, _enketo_preview_url)
-                f.write('%s : %s \n' % (id_string, data_value))
+
+        if use_username:
```

Comment:
why not put this conditional inside the loop to remove duplication?

Reply:
can you remove all the new duplicate by having the condition affect whether `username` is `None` or not and the value of `id_string`? that looks doable to me

Reply:
Moved the conditional inside the loop in the latest commit.

Also, I've modified this code block removing the duplicate code. Using a conditional to effect the value of the `xform_uuid` passed into the functions.